activesupport/lib/active_support/combined_configuration.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+require "active_support/core_ext/enumerable"
+
+module ActiveSupport
+  # Allows for configuration keys to be pulled from multiple backends. Keys are pulled in first-found order from
+  # the configuration backends that the combined configuration has been initialized with.
+  #
+  # This is used by Rails to offer a unified API for fetching credentials from both ENV and the encrypted file.
+  class CombinedConfiguration
+    def initialize(*configurations)
+      @configurations = configurations
+    end
+
+    # Combined interface for #[] and #dig. Examples:
+    #
+    #   creds.grab(:db_host) # => creds[:db_host]
+    #   creds.grab(:database, :host) # => creds.dig(:database, :host)
+    def grab(*keys)
```

Comment:
Shouldn't just this be `dig`? Since the behavior is identical, might as well use a well known name.